# Introduction {.unnumbered}

## What are damage patterns? {.unnumbered}

Damage patterns on ancient DNA molecules occur due to increased miscoding lesions at the end of molecules. When DNA molecules start to decompose (i.e., repair mechanisms are lost once an organism dies), the very long DNA molecules start to fragment due to 'nicks' occurring on the sugar-phosphate of one of the strands, weakening the structure and causing the molecule to cleave into two. However, this cleavage is not necessarily 'clean', i.e., occurs on both strands at the same position. Rather, when the two uneven 'nicks' cause the DNA molecule to cleave into two, this results in  a 'jagged' break - with the molecules having 'overhangs' of one strand being longer than the other of each of the new two now-'independent' molecules.

The resulting single-stranded overhangs leave the nitrogenous-bases 'exposed' on the overhang to the surrounding environment. It was found

It is important to note that the library construction method will influence damage, e.g., is the library constructed from double-stranded DNA or single-stranded DNA, is the polymerase in the initial library amplification proof-reading or not, and so on.

@fig-briggs-2007


![First reported misincorporation lesion 'smiley plot' from Neanderthal DNA [@Briggs2007-ao]. Reproduced here under free access.](assets/images/briggs_2007_original_Damage_pattern.jpeg){#fig-briggs-2007}

## How to generate damage patterns

There is a range of software that can generate damage pattern plots from ancient DNA NGS libraries. The vast majority of tools require to be of sequencing reads aligned to a reference genome. Here we make suggestions of some tools that you can use to generate such plots.

### Genomics

These tools general take BAM files as input (i.e., after mapping of FASTQ files to a reference genome using a short-read aligner )

- mapDamage
  - Source: [https://github.com/ginolhac/mapDamagee](https://github.com/ginolhac/mapDamage)
  - Documentation: [https://ginolhac.github.io/mapDamage](https://ginolhac.github.io/mapDamage)
  - Citation: [@Jonsson2013-zb]
- DamageProfiler
  - Source: [https://github.com/Integrative-Transcriptomics/DamageProfiler](https://github.com/Integrative-Transcriptomics/DamageProfiler)
  - Documentation: [https://damageprofiler.readthedocs.io/en/latest/](https://damageprofiler.readthedocs.io/en/latest/)
  - Citation: [@Neukamm2021-ul]

### Metagenomics

- MaltExtract
  - Source: [https://github.com/rhuebler/MaltExtract](https://github.com/rhuebler/MaltExtract)
  - Documentation:  [https://github.com/rhuebler/MaltExtract](https://github.com/rhuebler/MaltExtract)
  - Citation: [@Hubler2019-qw]