# Proofreading enzymes

![Sebastian Sprudelwasser, on his way to a Berlin Rave. By Anne Kathrine Wiborg Runge](assets/images/proofreading-polymerase/sebastian_sprudelwasser.png){#fig-proofreading-polymerase-caricature height=400px}

\newpage

```{r}
#| label: fig-template
#| fig-cap: Example of a smiley plot of a double stranded library amplified after adapter ligation with a proofreading NEBNext Ultra II Q5 Master polymerase. Data taken from library HH06D of [@Cai2022-rp]. Damage data generated using [DamageProfiler](/intro.qmd) and plotted using R and tidyverse packages [@Wickham2019-ot].
#| echo: false
#| warning: false
#| error: false

library(readr)
library(tidyr)
library(dplyr)
library(ggplot2)
library(patchwork)
source("assets/src/R/style.r")
source("assets/src/R/damage-profiler.r")
type_colours <- type_colours_dp

raw_5p <- readr::read_tsv("assets/data/proofreading-polymerase/5p_freq_misincorporations.txt", comment = "#")
raw_3p <- readr::read_tsv("assets/data/proofreading-polymerase/3p_freq_misincorporations.txt", comment = "#")

long_5p <- pivot_longer_freqmisincorporat(raw_5p, reverse = FALSE, type_colours = type_colours)
long_3p <- pivot_longer_freqmisincorporat(raw_3p, reverse = TRUE, type_colours = type_colours)

smiley_5p <- plot_longer_freqmisincorporat(long_5p, reverse = FALSE, type_colours)
smiley_3p <- plot_longer_freqmisincorporat(long_3p, reverse = TRUE, type_colours)

smiley_5p + smiley_3p

```

Sometimes the type of polymerase you use during library construction will influence the type of damage pattern you will receive.

In the example above, @Cai2022-rp found a funny smiley plot, where while the 3' G to A patterns look like a classic non-UDG (i.e. non-USER treated, thus retaining damage) plot, the frequency of C to T on the 5' end was extremely reduced.
This is because of the choice of polymerase used during library construction was not conductive to the nature of ancient DNA.

In this paper, the Q5Â® High-Fidelity DNA Polymerase was used, which has no tolerance for uracils.
During replication, it stalls on uracils in the template DNA, and hence cannot proceed with extending the newly forming complement strand. 
As the original strand with the uracil stops early, it results in a single-stranded overhang without an adapter.
Thus. the C-to-T deamination at the 5' termini of the reads will be absent in the plot.
In contrast, the original complementary DNA strand, which is filled-in by another DNA polymerase during the blunt-ending step of the double-stranded library preparation, will still contain the complementary A nucleotide.
This means the G-to-A misincorporation pattern is retained in 3' termini of the reads, even if the deaminated cytosines were lost on the first strand.

While having a low error rate is great for modern genomics, this can be less optimal for preserving ancient DNA damage for profiling later on.
Furthermore, you may have lost a fraction of true ancient DNA reads (those that ahve uracils in both strands).

In the case of this particular enzyme, it may not too much of a problem to prove authenticity visually as you retain the damage signal on the 3'.
However this may be problematic for downstream aDNA validation tools that may have an expected 'model' of ancient DNA damage.

:::{.callout-tip}
The choice of enzyme only matters during the first amplification after adapter ligation. At subsequent amplifications, the (misincorporated) thymines have already been integrated into the template molecules, so it doesn't matter which enzyme you use.
::: 
